evm/v0.5/contracts/dev/ServiceAgreementConsumer.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity 0.5.0;
+
+contract ServiceAgreementConsumer {
```

Comment:
This contract would have shared the same name with `evm/v0.5/contracts/tests/ServiceAgreementConsumer.sol`... but I renamed that contract to `BasicServiceAgreementConsumer`. Open to other naming conventions if this seems confusing.

Reply:
I think this is problematic because `Consumer` means two different things in the different contracts. Can we call this `ServiceAgreeementDecoder` and `OracleSignatureDecoder` instead, and leave the other one with the previous name?

Reply:
:+1: